cmd/consumer/main.go

```diff
@@ -40,8 +44,34 @@ func main() {
 	if err != nil {
 		logger.Fatal("Failed to load config", zap.Error(err))
 	}
+	qClient, err := queue.New(logger, ctx, &configuration.RabbitMq)
+	if err != nil {
+		logger.Fatal("Failed to instantiate queue client", zap.Error(err))
+	}
+	defer qClient.CloseConnection()
+
+	grpcClient, err := grpc.NewClient(configuration.GrpcClient.GrpcAddress, logger)
+	if err != nil {
+		logger.Fatal("Unable to create GRPC client.", zap.Error(err))
+	}
+	defer grpcClient.Close()
+	logger.Info("GRPC client connected to server")
+	var wg sync.WaitGroup
```

Comment:
Curious as to what will happen if you call `wg.Wait()` if this doesn't get set,  or the for loop doesn't run.

I tend to do `wg := sync.WaitGroup{}` so i know its always available.

Reply:
Interestingly a few tutorials don't explicitly instantiate the waitgroup.

Here is an example of a tutorial that doesn't: https://gobyexample.com/waitgroups

But will update to instantiate anyway